server/controllers/joinLeaveCancelController.js

```diff
@@ -0,0 +1,107 @@
+"use strict";
+
+const dbName = "ParkGames";
+const collectionEvents = "Events";
+const collectionParticipants = "Participants";
+const collectionUserEvents = "UserEvents";
+const collectionRooms = "Rooms";
+const assert = require("assert");
+var ObjectId = require("mongodb").ObjectID;
+//env vairables
+require("dotenv").config();
+const {
+  //handleJoinEvent.
+  getEventById,
+  getParticipantsById,
+  getMatchingParticipant,
+  addParticipant,
+  addUserEvent,
+  removeParticipant,
+  removeUserEvent,
+  //handleLeaveEvent.
+
+  //handleCancelEvent.
+} = require("../services/joinLeaveCancelService");
+
+//@endpoint POST /joinEvent
+//@desc join the event selected from ViewActivity component.
+//@access PRIVATE - will need to validate token? YES
+const handleJoinEvent = async (req, res, next) => {
+  const participantDetails = req.body.participantDetails;
+  const eventInformation = req.body.eventInformation;
+  try {
+    let getEvent = await getEventById(eventInformation._id);
+    //see if there is a participant ID in that event. If so then there are at least 1 participant.
+    //if there is a participant ID.
+    //check if that participant doesnt already exist... in that event.
+    let getParticipants = await getParticipantsById(getEvent.participantId);
+    //if you get participants. Which you will 100% because if you have a apeticipant ID then there are participants
+    if (getParticipants && getEvent) {
+      //check if any of the participants in the array match the current participant trying to join.
+      let existingParticipant = getMatchingParticipant(
+        getParticipants,
+        participantDetails.userId
+      );
+      //if they do match...
+      if (existingParticipant) {
+        res.status(400).json({
```

Comment:
This doesn't seem like the right status code to me, `400` indicates bad request parameters. This seems more like a `409` to me